### PR TITLE
Working egg incubation

### PR DESCRIPTION
**Pull request by friscoMad**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pogom/account.py [hunk 1]
```diff
@@ -494,6 +493,7 @@ def spinning_try(api, fort, step_location, account, map_dict, args):
             clear_inventory(api, account)
             account['session_spins'] += 1
             account['used_pokestops'][fort['id']] = time.time()
+            incubate_eggs(api, account)
```
Comment (friscoMad) [inline pogom/account.py:496]: Added incubate here for testing but not a bad place anyway

File: pogom/account.py [hunk 2]
```diff
@@ -523,6 +523,10 @@ def parse_inventory(api, account, map_dict):
     parsed_items = 0
     parsed_pokemons = 0
     parsed_eggs = 0
+    parsed_incubators = 0
+    # while not using deltas we are inserting all every time
+    account['incubators'] = []
```
Comment (friscoMad) [inline pogom/account.py:528]: This is now needed as we are using lists so the same incubator/egg can end in the list twice, when we start using deltas there shouldn't be needed if we reset inventory when we reset timestamp to 0.

File: pogom/account.py [hunk 3]
```diff
@@ -543,19 +547,24 @@ def parse_inventory(api, account, map_dict):
...
-                account['incubators'][incubator['id']] = {
-                    'item_id': incubator['item_id'],
-                    'uses_remaining': incubator.get('uses_remaining', 0),
-                    'pokemon_id': incubator.get('pokemon_id', 0)
-                }
-        if 'pokemon_data' in item_data:
+                if incubator.get('pokemon_id', 0):
+                    left = (incubator['target_km_walked']
+                            - account['walked'])
+                    log.info('Egg kms remaining: %.2f', left)
```
Comment (friscoMad) [inline pogom/account.py:553]: This was for testing but useful, maybe we can change to debug

File: pogom/account.py [hunk 4]
```diff
@@ -543,19 +547,24 @@ def parse_inventory(api, account, map_dict):
...
+                        'id': incubator['id'],
+                        'item_id': incubator['item_id'],
+                        'uses_remaining': incubator.get('uses_remaining', 0),
+                    })
+                    parsed_incubators += 1
+        elif ('pokemon_data' in item_data and
+              item_data['pokemon_data'].get('id', 0)):
             p_data = item_data['pokemon_data']
             p_id = p_data.get('id', 0)
-            pokemon_id = p_data.get('pokemon_id', 0)
```
Comment (friscoMad) [inline pogom/account.py:554]: Refactored this a bit, I think is is clearer now

File: pogom/account.py [hunk 5]
```diff
@@ -608,12 +618,10 @@ def cleanup_account_stats(account, pokestop_timeout):
...
-    used_pokestops = dict(account['used_pokestops'])
-    for pokestop_id in account['used_pokestops']:
-        last_attempt = account['used_pokestops'][pokestop_id]
-        if (last_attempt + pokestop_timeout) < time.time():
-            del used_pokestops[pokestop_id]
-    account['used_pokestops'] = used_pokestops
+    account['used_pokestops'] = {
+            k: v for k, v in account['used_pokestops'].iteritems()
+            if (v + pokestop_timeout) > time.time()
+        }
```
Comment (friscoMad) [inline pogom/account.py:624]: Even if not in use, also refactored the filter to use a comprehension, but un tested as it is not used.

File: pogom/account.py [hunk 6]
```diff
@@ -635,25 +643,19 @@ def clear_inventory_request(api, item_id, drop_count):
...
-
-    indexes = range(len(item_ids))
-    clear_responses = []
-    for i in indexes:
-        item_count = account['items'].get(item_ids[i], 0)
-        if item_count > random.randint(5, 10):
-            item_id = item_ids[i]
-            item_name = item_names[i]
-            drop_count = item_count - 5
+    items = [(1, 'Pokeball'), (2, 'Greatball'), (3, 'Ultraball'),
```
Comment (friscoMad) [inline pogom/account.py:646]: Easy to mantain and to iterate

File: pogom/account.py [hunk 7]
```diff
@@ -635,25 +643,19 @@ def clear_inventory_request(api, item_id, drop_count):
...
+             (201, 'Revive'), (202, 'Max Revive'),
+             (701, 'Razz Berry'), (703, 'Nanab Berry'), (705, 'Pinap Berry'),
+             (1101, 'Sun Stone'), (1102, 'Kings Rock'), (1103, 'Metal Coat'),
+             (1104, 'Dragon Scale'), (1105, 'Upgrade')]
+
+    for item_id, item_name in items:
+        item_count = account['items'].get(item_id, 0)
+        random_max = random.randint(5, 10)
+        if item_count > random_max:
+            drop_count = item_count - random_max
```
Comment (friscoMad) [inline pogom/account.py:658]: Previously always left 5 items now it leaves the random limit we used.

File: pogom/account.py [hunk 8]
```diff
@@ -681,36 +681,30 @@ def clear_inventory(api, account):
             else:
                 log.warning('Failed to clear inventory.')
 
-            log.debug('Recycled inventory: \n\r{}'.format(clear_responses))
+            log.debug('Recycled inventory: \n\r{}'.format(clear_result))
 
-    return clear_responses
+    return
```
Comment (friscoMad) [inline pogom/account.py:686]: The returns were really strange and clear_responses was only used inside the loop so it kept getting bigger, I changed to don't return anything and log each clean response.

Comment (Alderon86) [reply pogom/account.py:686]: blame @Obihoernchen ;D

Comment (Obihoernchen) [reply pogom/account.py:686]: (╯°□°）╯︵ ┻━┻

Because I did something else with it ;)

**Pull request merged**